### ISSUE TITLE
A function decorator for PyTorch modules, to export ONNX

### ISSUE THREAD

**Issue by dashesy** [pytorch/pytorch#20867]
## 🚀 Feature

I would like to have a decorator like

```python
    @onnx_convert('MeanVarianceNormalization')
    def forward(self, x):
        ....
```

To return correct `g.op(MeanVarianceNormalization)` to the tracer.

## Motivation

I have this `MVN` implementation:

```python
class MVN(nn.Module):
    """Mean-Variance Normalization (MVN) Layer
    """
    def __init__(self, normalize_variance=True, across_channels=False, eps=None):
        super(MVN, self).__init__()
        self.normalize_variance = normalize_variance
        self.across_channels = across_channels
        if eps is None:
            eps = 1e-9
        self.eps = eps

    def extra_repr(self):
        """Extra information
        """
        return "eps={}{}{}".format(
                self.eps,
                ", normalize_variance" if self.normalize_variance else "",
               ", across_channels" if self.across_channels else "",
        )

    def forward(self, x):
        std = None
        shape = x.data.shape
        n_b = shape[0]
        if self.across_channels:
            if shape[0] == 1:
                # single batch normalization reduction
                mean = x.mean()
                if self.normalize_variance:
                    std = x.std()
            else:
                mean = x.view(n_b, -1).mean(1).view(n_b, 1, 1, 1)
                if self.normalize_variance:
                    std = x.view(n_b, -1).std(1).view(n_b, 1, 1, 1)
        else:
            n_c = shape[1]
            mean = std = x.view(n_b, n_c, -1).mean(2).view(n_b, n_c, 1, 1)
            if self.normalize_variance:
                std = x.view(n_b, n_c, -1).std(2).view(n_b, n_c, 1, 1)
        if std is not None:
            return (x - mean) / (std + self.eps)
        return x - mean
```

I cannot export it because [std is not supported in ONNX](https://github.com/onnx/onnx/issues/2045) but ONNX already has [MeanVarianceNormalization](https://github.com/onnx/onnx/blob/master/docs/Operators.md#MeanVarianceNormalization) that I would like to be used with this `nn.Module`. 

## Pitch

Instead of having an exporter that knows every detail about everything, let the `nn.Module` (or `torch.autograd.Function`) implementer provide a function that returns the right export ops.

## Alternatives

1. Modify symbolic.py (not a good idea because it may change in the next update)
2. monkey-patch symbolic.py, this file already monke-patches using `import torch.onnx.utils` so I hope this works
3. provide decorator as `Functional` level instead of `nn.Module`, a little more cumbersome in simple cases but not bad

## Additional context

I also need this functionality for custom `nn.Modules` some of which using `CUDAExtension` or `CppExtension`, I would like to use this functionality to automatically export ONNX from full model that includes these custom layer/ops

I first reported this [here](https://github.com/microsoft/onnxruntime/issues/1087) but it is really related to PyTorch's ONNX exporter.

**Comment by mrshenli** [comment]
cc @houseroad

**Comment by garymm** [comment]
You can provide a custom symbolic for any [aten op](https://pytorch.org/docs/master/onnx.html#torch.onnx.register_custom_op_symbolic) or [your own torch.autograd.Function](https://pytorch.org/docs/master/onnx.html#torch-autograd-functions).

@dashesy please explain if these are not sufficient for your purposes.

**Comment by Bea07** [comment]
@dashesy  Have you solved this problem? I have similar issue.
What is the difference between Instance normalization and MVN in onnx? 
My pythorch model is currently using instance normalization but my ultimate scope is to convert the model in Openvino passing through onnx, but there is an incompatibility with this normalization and I need to use MVN instead (compatibility operations onnx-openvino: https://docs.openvino.ai/2020.3/_docs_MO_DG_prepare_model_Supported_Frameworks_Layers.html#onnx_supported_operators)
